spec/services/preserved_object_handler_spec.rb

```diff
@@ -110,4 +110,25 @@
       po_handler.create_after_validation
     end
   end
+
+  describe 'endpoint validation' do
+    it 'errors when endpoint is not an Endpoint object' do
+      poh = described_class.new(druid, incoming_version, incoming_size, 1)
+      expect(poh).to be_invalid
+      expect(poh.errors.messages).to include(endpoint: ["must be an actual Endpoint"])
+    end
+    it 'errors when endpoint_type is not online' do
+      endpoint_type = EndpointType.create(type_name: 'foo', endpoint_class: 'archive')
+      archival_endpoint = Endpoint.create(endpoint_name: 'archive1',
+                                          endpoint_type_id: endpoint_type.id,
+                                          endpoint_node: 'node',
+                                          storage_location: 'somewhere')
```

Comment:
You could replace both these 5 lines with one factory invocation.  We have one just for archival endpoints.
```ruby
archival_endpoint = create(:archive_endpoint)
```

Reply:
https://github.com/sul-dlss/preservation_catalog/pull/873